orangecontrib/text/pubmed.py

```diff
@@ -0,0 +1,480 @@
+    except:
+        return None
+
+
+def _date_to_iso(date):
+    possible_date_formats = [
+        '%Y %b %d',
+        '%Y %b',
+        '%Y',
+    ]
+
+    season_mapping = {
+        'fall': 'Sep',
+        'winter': 'Dec',
+        'spring': 'Mar',
+        'summer': 'Jun',
+    }
+
+    for date_format in possible_date_formats:
+        try:
+            return datetime.strptime(date, date_format).date().isoformat()
+        except Exception:
+            continue  # Try the next format.
+
+    date = date.lower()
+    # Seasons to their respective months.
+    for season, month in season_mapping.items():
+        date = date.replace(season, month)
+    try:  # Try month range (e.g. Jan-Mar).
+        date = date.split('-')[0]  # 2015 Sep-Dec --> 2015 Sep
+        return datetime.strptime(
+                date,
+                possible_date_formats[1]
+        ).date().isoformat()
+    except Exception:
+        warnings.warn(
+                'Could not parse "{}" into a date.'.format(
+                    date
+                ),
+                RuntimeWarning
+        )
+        return None
+
+
+def _records_to_corpus_entries(records, includes_metadata):
+    """Create corpus entries from records.
+
+    Args:
+        records (list): A list of dictionaries that hold record data.
+        includes_metadata (list): A list of tuples, where the
+            elements hold the names and tags of the metadata fields that we
+            wish to extract from the xml.
+
+    Returns:
+        list, list: Metadata and class values. Metadata is an array of size
+            n*m, where n is the number of article records contained within
+            'data' and m is the number of metadata fields we've chosen to
+            extract from the xml. The variable class_values is a list, where
+            the elements are class values for each article instance.
+    """
+    class_values = []
+    metadata = np.empty((len(records), len(includes_metadata)), dtype=object)
+
+    for row_num, record in enumerate(records):
+        fields = [
+                _join_if_list(record.get(field_key, ''))
+                for field_name, field_key
+                in includes_metadata
+            ]
+        fields[-1] = _date_to_iso(fields[-1])
+        metadata[row_num] = np.array(fields)[None, :]
+
+        class_values.append(_mesh_headings_to_class(record.get('MH')))
+    return metadata, class_values
+
+
+def _corpus_from_records(records, includes_metadata):
+    """Receives PubMed records and transforms them into a corpus.
+
+    Args:
+        records (list): A list of PubMed entries.
+        includes_metadata (list): A list of text fields to include.
+
+    Returns:
+        corpus: The output Corpus.
+    """
+    meta_values, class_values = _records_to_corpus_entries(
+            records,
+            includes_metadata=includes_metadata
+    )
+    meta_vars = [
+        StringVariable.make(field_name)
+        for field_name, field_key
```

Comment:
Don't know about `field_key`, but ISO timestamp should be put into a TimeVariable.